File: CSC 120/workspace/VuProject2/deck.py
import card
import random

SMALLEST_CARD_NUMBER = 2
BIGGEST_CARD_NUMBER = 10
NUM_CARDS_ON_TWO_HANDS = 10


class Deck:

    def __init__(self):
        """
        create a deck of 52 cards as a list
        """
        self.__deck = []
        suits = ['C', 'D', 'H', 'S']
        rank = [str(i) for i in range(SMALLEST_CARD_NUMBER, BIGGEST_CARD_NUMBER + 1)]
        rank.extend(['J', 'Q', 'K', 'Ace'])

        for suit in suits:
            for j in range(len(rank)):
                my_card = card.Card(rank[j], suit, j + 2)

                self.__deck.append(my_card)

    def shuffle_deck(self):
        """

        :return: randomly shuffle the deck
        """

        random.shuffle(self.__deck)

    def deal_a_card(self):
        """

        :return: deal the first card
        """
        return self.__deck.pop(0)

    def get_length(self):
        """

        :return: the length of deck
        """
        return len(self.__deck)

    def __str__(self):
        """

        :return: print out deck of cards
        """
        return str(self.__deck)

    def __repr__(self):
        """

        :return: print out deck of cards with better representation
        """
        return str(self.__deck)

File: CSC 120/workspace/IdeaProjects/src/project3/Deck.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Arrays;
import java.util.List;

/**
 * This class is to make a deck of card
 * topIndex is the index of a card in the deck to be dealt
 */
public class Deck{

    private ArrayList<Card> listCards;
    private int topIndex = 0;

    /**
     * Create a shuffled deck as an array list of 52 cards
     */
    public Deck(){
        listCards = new ArrayList<Card>();
        ArrayList<String> suits = new ArrayList<String>();
        suits.add("S");
        suits.add("C");
        suits.add("D");
        suits.add("H");
        String[] ranks = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "Ace"};
        List<String> allRanks = Arrays.asList(ranks);
        for (Object suit : suits){
            for (Object rank : allRanks){
                int indexRank = allRanks.indexOf(rank);
                Card my_card = new Card((String) rank, (String) suit, indexRank+2);
                listCards.add(my_card);
            }
        }
        shuffleDeck();
    }

    /**
     *
     * @return a deck as a string list with better representation
     */
    public String toString(){
        return listCards.toString();
    }

    /**
     * randomly reorganize the order of the cards in a deck
     */
    private void shuffleDeck(){
        Collections.shuffle(listCards);

    }

    /**
     *
     * @return get the size of deck after dealing a card
     * The size decrements by 1 as the topIndex of the card increments by 1
     * The current size = the initial size (52) - the topIndex as it += 1
     * The method doesn't affect the number of cards in the original deck (which is still 52)
     */
    public int getSizeDeck(){
        return listCards.size() - topIndex;
    }

    /**
     *
     * @return a card dealt at its topIndex and the topIndex is constantly updated (increments by 1)
     */
    public Card dealCard(){
        Card dealtCard = listCards.get(topIndex);
        topIndex = topIndex + 1;
        return dealtCard;

    }




}